internal/controllers/provisioningrequest_upgrade.go

```diff
@@ -0,0 +1,183 @@
+		return false, nil
+	}
+
+	managedCluster := &clusterv1.ManagedCluster{}
+	err = t.client.Get(ctx, types.NamespacedName{Name: managedClusterName}, managedCluster)
+	if err != nil {
+		return false, fmt.Errorf("failed to get ManagedCluster: %w", err)
+	}
+
+	templateReleaseVersion, err := semver.NewVersion(template.Spec.Release)
+	if err != nil {
+		return false, fmt.Errorf("failed to parse template version: %w", err)
+	}
+	managedClusterVersion, err := semver.NewVersion(managedCluster.GetLabels()["openshiftVersion"])
+	if err != nil {
+		return false, fmt.Errorf("failed to parse ManagedCluster version: %w", err)
+	}
+	cmp := templateReleaseVersion.Compare(*managedClusterVersion)
+	if cmp == 1 {
+		return true, nil
+	} else if cmp == -1 {
+		return false, fmt.Errorf("template version (%v) is lower then ManagedCluster version (%v), no upgrade requested",
+			templateReleaseVersion, managedClusterVersion)
+	}
+	return false, nil
+}
+
+func (t *provisioningRequestReconcilerTask) handleUpgrade(
+	ctx context.Context, renderedClusterInstance *siteconfig.ClusterInstance) (ctrl.Result, error) {
+	t.logger.InfoContext(
+		ctx,
+		"Start handleUpgrade",
+	)
+	clusterTemplate, err := t.getCrClusterTemplateRef(ctx)
+	if err != nil {
+		return requeueWithError(fmt.Errorf("failed to get clusterTemplate: %w", err))
+	}
+
+	ibgu := &ibgu.ImageBasedGroupUpgrade{}
+	err = t.client.Get(ctx, types.NamespacedName{Name: t.object.Name, Namespace: renderedClusterInstance.Namespace}, ibgu)
+	if err != nil && errors.IsNotFound(err) {
+		ibgu, err = utils.GetIBGUFromUpgradeDefaultsConfigmap(
+			ctx, t.client, clusterTemplate.Spec.Templates.UpgradeDefaults,
+			clusterTemplate.Namespace, utils.UpgradeDefaultsConfigmapKey,
+			renderedClusterInstance.Spec.ClusterName, t.object.Name, renderedClusterInstance.Namespace)
+		if err != nil {
+			return requeueWithError(fmt.Errorf("failed to generate IBGU for cluster: %w", err))
+		}
+		if err := utils.CreateK8sCR(ctx, t.client, ibgu, t.object, utils.UPDATE); err != nil {
+			return requeueWithError(fmt.Errorf("failed to create IBGU: %w", err))
+		}
+
+		t.logger.InfoContext(
+			ctx,
+			fmt.Sprintf(
+				"Upgrade initiated. Created IBGU %s in the namespace %s",
+				ibgu.GetName(),
+				ibgu.GetNamespace(),
+			),
+		)
+
+		utils.SetStatusCondition(&t.object.Status.Conditions,
+			utils.PRconditionTypes.UpgradeCompleted,
+			utils.CRconditionReasons.InProgress,
+			metav1.ConditionFalse,
+			"Upgrade is initiated",
+		)
+		if err := utils.UpdateK8sCRStatus(ctx, t.client, t.object); err != nil {
+			return requeueWithError(fmt.Errorf("failed to update ClusterRequest CR status: %w", err))
+		}
+
+	} else if err != nil {
+		return requeueWithError(fmt.Errorf("error getting IBGU: %w", err))
+	}
+
+	if isIBGUProgressing(ibgu) {
+		utils.SetStatusCondition(&t.object.Status.Conditions,
+			utils.PRconditionTypes.UpgradeCompleted,
+			utils.CRconditionReasons.InProgress,
+			metav1.ConditionFalse,
+			"Upgrade is in progress",
+		)
```

Comment:
I think `provisioningStatus.state` should be updated to progressing with a detailed message to align with the current process? This can be done in a follow-up PR.

Reply:
Do we still need a separate upgrade condition then? I asked about this very early on and the sense I got was a separate condition was more preferred.

Reply:
What I meant is just for the overall status, when upgrade condition is changed to InProgress, we would update the overall provisioningStatus to progressing as well, same for the completion.